backend/app/categories/controller.py

```diff
@@ -68,11 +69,32 @@ async def add_item_route(category_id: CategoryEnum, item: Union[Movie, Anime, Sh
 
 
 @router.patch("/{category_id}/{item_id}", response_model=ItemInResponse, dependencies=[Depends(verify_token)], tags=["update"])
-async def update_item_route(category_id: CategoryEnum, item_id: ObjectID,
-                            db: AsyncIOMotorClient = Depends(get_database)) -> ItemInResponse:
+async def update_item_route(category_id: CategoryEnum,item_id: ObjectID, 
+                            item: Union[Movie, Anime, Show, Music, Book],db: AsyncIOMotorClient = Depends(get_database)) -> ItemInResponse:
     """Update a given item"""
-    # TODO add update route
-    return {"success": True}
+    classof = {
+        CategoryEnum.movies: Movie,
+        CategoryEnum.shows: Show,
+        CategoryEnum.anime: Anime,
+        CategoryEnum.books: Book,
+        CategoryEnum.music: Music
+    }
+
+    # checking if request body is of correct type
+    if not isinstance(item, classof[category_id]):
+        raise HTTPException(
+            status_code=400, detail=f'Item does not match {category_id} schema')
+    if (ObjectId.is_valid(item_id)==False):
+        raise HTTPException(
+            status_code=400, detail='Not a valid ObjectID')
+    try:
+        newvalues = {"$set":item.dict()}
+        _res = await db[category_id]["data"].update_one({"_id": item_id},newvalues)
+        updated_item = await db[category_id]["data"].find_one({"_id": item_id}) 
```

Comment:
Since you have just updated it, you shouldn't fetch it again, just return the `item.dict()` to the user. It's inefficient to make two database calls per update when we actually only need one.